### ISSUE TITLE
Metrics page not using HTTPS after TLS enabled

### ISSUE THREAD

**Issue by AstroProfundis** [tikv/tikv#5340]
## Bug Report
When creating a TiDB cluster with TLS support enabled (on server side), the metrics page of TiKV still uses plain HTTP to serve connections, while TiDB and PD are using HTTPS, so Prometheus can't scrape metrics in an unique way.

### What version of TiKV are you using?

```
Release Version:   3.0.2
Git Commit Hash:   a65d10c6f5fb62d3aafcab6cfdcd5b36c2b33d35
Git Commit Branch: HEAD
UTC Build Time:    2019-08-07 02:31:51
Rust Version:      rustc 1.37.0-nightly (0e4a56b4b 2019-06-13)
```

### What operating system and CPU are you using?

Running in a Kubernetes cluster, but that doesn't matter.

### Steps to reproduce

Create a cluster with TLS enabled, and try to access each node's `/metrics` page.

TiDB:
```
# curl -Ik --cacert ./ca.crt --cert ./client.crt --key ./client.key https://10.233.48.147:10080/metrics
HTTP/1.1 200 OK
Content-Length: 244902
Content-Type: text/plain; version=0.0.4; charset=utf-8
Date: Tue, 27 Aug 2019 03:59:56 GMT
```

PD:
```
# curl -Ik --cacert ./ca.crt --cert ./client.crt --key ./client.key https://10.233.61.189:2379/metrics
HTTP/1.1 200 OK
Access-Control-Allow-Headers: accept, content-type, authorization
Access-Control-Allow-Methods: POST, GET, OPTIONS, PUT, DELETE
Access-Control-Allow-Origin: *
Content-Length: 252488
Content-Type: text/plain; version=0.0.4; charset=utf-8
Date: Tue, 27 Aug 2019 04:00:33 GMT
```

TiKV:
```
# curl -kv --cacert ./ca.crt --cert ./client.crt --key ./client.key https://10.233.77.255:20180/metrics
* About to connect() to 10.233.77.255 port 20180 (#0)
*   Trying 10.233.77.255...
* Connected to 10.233.77.255 (10.233.77.255) port 20180 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
* NSS error -12263 (SSL_ERROR_RX_RECORD_TOO_LONG)
* SSL received a record that exceeded the maximum permissible length.
* Closing connection 0
curl: (35) SSL received a record that exceeded the maximum permissible length.
```

TiKV with HTTP:
```
# curl -v http://10.233.77.255:20180/metrics
* About to connect() to 10.233.77.255 port 20180 (#0)
*   Trying 10.233.77.255...
* Connected to 10.233.77.255 (10.233.77.255) port 20180 (#0)
> GET /metrics HTTP/1.1
> User-Agent: curl/7.29.0
> Host: 10.233.77.255:20180
> Accept: */*
> 
< HTTP/1.1 200 OK
< content-length: 291021
< date: Tue, 27 Aug 2019 04:06:42 GMT
< 
{ [data not shown]
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 521.56
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 1048576
....
```

### What did you expect?
All requests are served as HTTPS when TLS is enabled, and client certificate should be validated, the same as TiDB/PD's behavior.

### What did happened?
TiKV doesn't use HTTPS for metrics page on TLS cluster.

**Comment by breeswish** [comment]
Thanks for your report. This is indeed a missing part.

**Comment by breeswish** [comment]
@rleungx Considering that only this metric page is not TLS, this unimplemented feature can be also regarded as a bug. Would you like to fix it (and cherry pick to 3.0 and 2.1)?
